lib/beaker/options/parser.rb

```diff
@@ -215,6 +221,37 @@ def parse_args(args = ARGV)
         @options
       end
 
+      # Parse hosts options from host files into a host options hash
+      #
+      # @return [Hash] Host options, containing all host-specific details
+      # @return [nil, File] generated hosts file reference, or nil if no
+      #   hosts file generated
+      # @raise [ArgumentError] if a hosts file is generated, but it can't
+      #   be read by the HostsFileParser
+      def parse_hosts_options
+        #read the hosts file that contains the node configuration and hypervisor info
+        hosts_options = Beaker::Options::HostsFileParser.parse_hosts_file(@options[:hosts_file])
+        return hosts_options, nil
+      rescue Errno::ENOENT
+        require 'tempfile'
+        require 'beaker-hostgenerator'
+        bhg_cli = BeakerHostGenerator::CLI.new( [ @options[:hosts_file] ] )
+        hosts_file_content = bhg_cli.execute
+        hosts_file = Tempfile.open( 'hosts.yml' )
+        begin
+          hosts_file.write( hosts_file_content )
```

Comment:
yeah, what @nwolfe said. I don't really understand what `:hosts_file_generated` options is doing, in general.  but also because it appears that we're always writing a `./hosts.yml` here.  current dir seems like a non-great place, but in addition i'd expect beaker to just write to the preserved_hosts file (created another ticket for earlier serialization of preserved_hosts: https://tickets.puppetlabs.com/browse/BKR-828).